### ISSUE TITLE
Passing command line arguments to a test

### ISSUE THREAD

**Issue by goriol** [tapjs/tapjs#273]
I use node-tap to check "code" quality of Cucumber features that are stored in a Git repository.
Two parameters are necessary to perform this check: the repo _location_ and the _branch_ to check (they vary from one developer to another).

I am looking for the best way to pass those arguments to the test.

I can make it work using something like `node test/features.js --repo=... branch=...`
Unfortunately the output is a bit messy: I can easily see one test failed but I have to scroll the output for the failed test to see _what_ failed (which is painfull as more than 2000 test are performed).

The _spec_ reporter is great to get a summary of the failed tests but is only available when using the tap command. However this command (`tap test/feature.js --reporter=spec`) doesn't allow for passing arguments to the test itself.

What approch would you suggest in this situation?

**Comment by isaacs** [comment]
In the very short term, you could use environment variables, since these are passed down to each of the test scripts by default.

As a new feature, I could see some value in something like `--test-args="foo --bar=bloo -quUx"` or something.  Would that work for you?

**Comment by isaacs** [comment]
Or, actually, scratch that.  Argv is an array, not a string.  I'd follow the pattern of `--node-arg` and `--nyc-arg` and have a `--test-arg` option that you could set multiple times, so for the above example, it'd be something like `tap test/*.js --test=arg=foo --test-arg=--bar=bloo --test-arg=-quUx`

**Comment by goriol** [comment]
Your last proposal is exactly what I need!
Thanks a lot.

In the short term, I'll use something like this:

```
// wrapper.js
var testArgs = process.argv.slice(2),
    reporter = ..., // option from the wrapper.js arguments
    tap = require('tap'),
    TMR = require('tap-mocha-reporter');

tap.unpipe(process.stdout);
tap.pipe(new TMR(reporter));
tap.spawn(process.execPath, ['actualTest.js'].concat(testArgs));

```

**Comment by isaacs** [comment]
Okie dokie, landed in 6.2.0.
